### PR TITLE
br: add region range check for batch scan regions

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?
Currently, the implement of pd's batch scan regions can't guarantee consistency, so we add extra check for return returned region [startKey, endKey) check to ensure if there are potential range gap, the method will return an error.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Check List 

Tests 

- [x] Unit test
- [x] Integration test

Side effects

Documentation

### Release note 

```release-note
None
```

### CODE REVIEW

File: br/pkg/restore/split.go [hunk 1]
```diff
@@ -316,13 +317,36 @@ func PaginateScanRegion(
...
+		if len(scanStartKey) == 0 ||
+			(len(endKey) > 0 && bytes.Compare(scanStartKey, endKey) >= 0) {
 			// All key space have scanned
 			break
 		}
 	}
+
+	// current pd can't guarantee the consistency of returned regions
+	if len(regions) == 0 {
+		return nil, errors.New("scan region return empty result")
```
Comment (YuJuncen) [inline br/pkg/restore/split.go:330]: maybe a normalized error (i.e. `berror.Errxxx`) here?

File: br/pkg/restore/util_test.go [hunk 2]
```diff
@@ -225,8 +225,7 @@ func (s *testRestoreUtilSuite) TestPaginateScanRegion(c *C) {
 	regionMap := make(map[uint64]*restore.RegionInfo)
 	regions := []*restore.RegionInfo{}
 	batch, err := restore.PaginateScanRegion(ctx, NewTestClient(stores, regionMap, 0), []byte{}, []byte{}, 3)
-	c.Assert(err, IsNil)
-	c.Assert(batch, DeepEquals, regions)
+	c.Assert(err, ErrorMatches, ".*scan region return empty result.*")
```
Comment (YuJuncen) [inline br/pkg/restore/util_test.go:228]: Maybe a case testing holes found during scanning regions?

**Review by YuJuncen — COMMENTED** [review]
rest lgtm

**Review by YuJuncen — APPROVED** [review]

File: br/pkg/restore/util_test.go (hunk 2 above)
Comment (glorv) [reply br/pkg/restore/util_test.go:228]: done

**Review by 3pointer — APPROVED** [review]

**Pull request merged**